### PR TITLE
Finalise theme switching migration for App

### PR DESCRIPTION
**Pull request by chrispader**
@grgia

### Details

Fixes last inconsistencies with theme switching and fixes minor issues after migration.

### Fixed Issues

$ https://github.com/Expensify/App/issues/32257
PROPOSAL: 

### Tests

- [x] Verify that no errors appear in the JS console

### QA Steps

- [x] Verify that no errors appear in the JS console

### PR Author Checklist

- [x] I linked the correct issue in the `### Fixed Issues` section above
- [x] I wrote clear testing steps that cover the changes made in this PR
    - [x] I added steps for local testing in the `Tests` section
    - [x] I added steps for the expected offline behavior in the `Offline steps` section
    - [x] I added steps for Staging and/or Production testing in the `QA steps` section
    - [x] I added steps to cover failure scenarios (i.e. verify an input displays the correct error message if the entered data is not correct)
    - [x] I turned off my network connection and tested it while offline to ensure it matches the expected behavior (i.e. verify the default avatar icon is displayed if app is offline)
    - [x] I tested this PR with a [High Traffic account](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#high-traffic-accounts) against the staging or production API to ensure there are no regressions (e.g. long loading states that impact usability).
- [x] I included screenshots or videos for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [x] I ran the tests on **all platforms** & verified they passed on:
    - [x] Android: Native
    - [x] Android: mWeb Chrome
    - [x] iOS: Native
    - [x] iOS: mWeb Safari
    - [x] MacOS: Chrome / Safari
    - [x] MacOS: Desktop
- [x] I verified there are no console errors (if there's a console error not related to the PR, report it or open an issue for it to be fixed)
- [x] I followed proper code patterns (see [Reviewing the code](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md#reviewing-the-code))
    - [x] I verified that any callback methods that were added or modified are named for what the method does and never what callback they handle (i.e. `toggleReport` and not `onIconClick`)
    - [x] I verified that the left part of a conditional rendering a React component is a boolean and NOT a string, e.g. `myBool && <MyComponent />`.
    - [x] I verified that comments were added to code that is not self explanatory
    - [x] I verified that any new or modified comments were clear, correct English, and explained "why" the code was doing something instead of only explaining "what" the code was doing.
    - [x] I verified any copy / text shown in the product is localized by adding it to `src/languages/*` files and using the [translation method](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60)
      - [x] If any non-english text was added/modified, I verified the translation was requested/reviewed in #expensify-open-source and it was approved by an internal Expensify engineer. Link to Slack message:
    - [x] I verified all numbers, amounts, dates and phone numbers shown in the product are using the [localization methods](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60-L68)
    - [x] I verified any copy / text that was added to the app is grammatically correct in English. It adheres to proper capitalization guidelines (note: only the first word of header/labels should be capitalized), and is approved by marketing by adding the `Waiting for Copy` label for a copy review on the original GH to get the correct copy.
    - [x] I verified proper file naming conventions were followed for any new files or renamed files. All non-platform specific files are named after what they export and are not named "index.js". All platform-specific files are named for the platform the code supports as outlined in the README.
    - [x] I verified the JSDocs style guidelines (in [`STYLE.md`](https://github.com/Expensify/App/blob/main/contributingGuides/STYLE.md#jsdocs)) were followed
- [x] If a new code pattern is added I verified it was agreed to be used by multiple Expensify engineers
- [x] I followed the guidelines as stated in the [Review Guidelines](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md)
- [x] I tested other components that can be impacted by my changes (i.e. if the PR modifies a shared library or component like `Avatar`, I verified the components using `Avatar` are working as expected)
- [x] I verified all code is DRY (the PR doesn't include any logic written more than once, with the exception of tests)
- [x] I verified any variables that can be defined as constants (ie. in CONST.js or at the top of the file that uses the constant) are defined as such
- [x] I verified that if a function's arguments changed that all usages have also been updated correctly
- [x] If any new file was added I verified that:
    - [x] The file has a description of what it does and/or why is needed at the top of the file if the code is not self explanatory
- [x] If a new CSS style is added I verified that:
    - [x] A similar style doesn't already exist
    - [x] The style can't be created with an existing [StyleUtils](https://github.com/Expensify/App/blob/main/src/styles/StyleUtils.js) function (i.e. `StyleUtils.getBackgroundAndBorderStyle(themeColors.componentBG)`)
- [x] If the PR modifies code that runs when editing or sending messages, I tested and verified there is no unexpected behavior for all supported markdown - URLs, single line code, code blocks, quotes, headings, bold, strikethrough, and italic.
- [x] If the PR modifies a generic component, I tested and verified that those changes do not break usages of that component in the rest of the App (i.e. if a shared library or component like `Avatar` is modified, I verified that `Avatar` is working as expected in all cases)
- [x] If the PR modifies a component related to any of the existing Storybook stories, I tested and verified all stories for that component are still working as expected.
- [x] If the PR modifies a component or page that can be accessed by a direct deeplink, I verified that the code functions as expected when the deeplink is used - from a logged in and logged out account.
- [x] If a new page is added, I verified it's using the `ScrollView` component to make it scrollable when more elements are added to the page.
- [x] If the `main` branch was merged into this PR after a review, I tested again and verified the outcome was still expected according to the `Test` steps.

### Screenshots/Videos
<details>
<summary>Android: Native</summary>

</details>

<details>
<summary>Android: mWeb Chrome</summary>

</details>

<details>
<summary>iOS: Native</summary>

</details>

<details>
<summary>iOS: mWeb Safari</summary>

</details>

<details>
<summary>MacOS: Chrome / Safari</summary>

</details>

<details>
<summary>MacOS: Desktop</summary>

</details>

No screenshots provided.

### CODE REVIEW

**Comment by grgia** [comment]
@parasharrajat testing for no errors on all platforms for both light and dark theme. We're not testing the theme itself or colors, just for errors

**Comment by grgia** [comment]
@parasharrajat you can use these

```
Onyx.merge("preferredTheme", 'light');


Onyx.merge("preferredTheme", 'dark');

```

**Review by chrispader — COMMENTED** [review]
This PR mostly just contains fixes for inconsistencies as well as a fix for [this bug](https://github.com/Expensify/App/pull/32264#issuecomment-1834305314).

Most importantly, `theme` should always be the first parameter in style util functions.

Removing this snippet in `index.html`, since it caused nested scroll bars in `react-native-pdf` to appear in dark mode when light mode is activated:

```css
:root {
     color-scheme: dark !important;
}
```

**Review by parasharrajat — CHANGES_REQUESTED** [review]
Screenshot 2023-12-01 at 12 16 07 AM

**Comment by chrispader** [comment]
> Screenshot 2023-12-01 at 12 16 07 AM

ah sorry, fixed!

**Comment by chrispader** [comment]
Should we stick with the `import type` approach or drop this? @grgia @parasharrajat

**Comment by chrispader** [comment]
> > Should we stick with the `import type` approach or drop this? @grgia @parasharrajat
> 
> This question is better suited for Slack.

Asked here: https://expensify.slack.com/archives/C01GTK53T8Q/p1701371123201489

File: src/components/Icon/index.tsx [hunk 1]
```diff
@@ -1,7 +1,7 @@
 import React, {PureComponent} from 'react';
 import {StyleProp, View, ViewStyle} from 'react-native';
 import withTheme, {ThemeProps} from '@components/withTheme';
-import withThemeStyles, {ThemeStylesProps} from '@components/withThemeStyles';
+import withThemeStyles, {type ThemeStylesProps} from '@components/withThemeStyles';
```
[the remark this answers is not in the document]
Comment (chrispader) [reply src/components/Icon/index.tsx:4]: Yes you can indeed mix it. Some imports can be variables, others types. TS will automatically handle this

**Comment by chrispader** [comment]
> Should we stick with the `import type` approach or drop this? @grgia @parasharrajat

Looks like we're going for this approach. Going to open a PR tmrw to standardize this approach, so think we can keep it like this in the PR :)

https://expensify.slack.com/archives/C01GTK53T8Q/p1701371741313369?thread_ts=1701371123.201489&cid=C01GTK53T8Q

**Review by grgia — DISMISSED** [review]

File: web/index.html [hunk 2]
```diff
@@ -31,9 +31,6 @@
         #root > div > div {
             height: 100% !important;
         }
-        :root {
-            color-scheme: dark !important;
-        }
```
[the remark this answers is not in the document]
Comment (grgia) [reply web/index.html:36]: it tests well for me with both modes and both system modes- do you have an example case?

[the remark this answers is not in the document]
Comment (grgia) [reply web/index.html:36]: I do kind of remember this issue when we implemented dark mode - but I don't think this fix works now that we support two themes

[the remark this answers is not in the document]
Comment (chrispader) [reply web/index.html:36]: Also, if we don't remove it, the scrollbar will always be dark inside PDFs, so just the other way around...

[the remark this answers is not in the document]
Comment (chrispader) [reply web/index.html:36]: We're already doing that here: https://github.com/Expensify/App/blob/ee7d5dc8d8c6d8706564e1dee685b093f9f4baa3/src/components/ColorSchemeWrapper/index.tsx

[the remark this answers is not in the document]
Comment (chrispader) [reply web/index.html:36]: Fixed:

https://github.com/Expensify/App/assets/20173411/fca9c336-9c46-445b-b7fe-0881a77d814b

[the remark this answers is not in the document]
Comment (chrispader) [reply web/index.html:36]: > There might be more elements like Context menus, tooltips, etc.

But none of these should include scrollable content right?

File: src/components/Modal/BaseModal.tsx [hunk 3]
```diff
@@ -48,6 +48,8 @@ function BaseModal(
 
     const safeAreaInsets = useSafeAreaInsets();
 
+    console.log({safeAreaInsets});
+
```
Comment (grgia) [inline src/components/Modal/BaseModal.tsx:52]: Let's remove console for lint :)

File: src/components/Modal/BaseModal.tsx [hunk 4]
```diff
@@ -203,12 +204,14 @@ function BaseModal(
             onLayout={onLayout}
             avoidKeyboard={avoidKeyboard}
         >
-            <View
-                style={[styles.defaultModalContainer, modalContainerStyle, modalPaddingStyles, !isVisible && styles.pointerEventsNone]}
-                ref={ref}
-            >
-                {children}
-            </View>
+            <ColorSchemeWrapper>
```
[the remark this answers is not in the document]
Comment (chrispader) [reply src/components/Modal/BaseModal.tsx]: This is fixed. I moved `ColorSchemeWrapper` inside the container `View`.

**Review by parasharrajat — CHANGES_REQUESTED** [review]
⬆️

**Comment by chrispader** [comment]
> Seeing this crash on the Web. Screenshot 2023-12-01 at 4 50 13 PM We should also refactor this function to pass the theme as the first parameter.

The function takes an object, so the order of the parameters doesn't matter here. I can move it to the first object property in code anyway, to make it clear :)

**Comment by chrispader** [comment]
Here you can see, the `MultipleAvatars` component is rendering without any problems

Screenshot 2023-12-02 at 08 00 25

**Review by parasharrajat — CHANGES_REQUESTED** [review]
https://github.com/Expensify/App/pull/32259#discussion_r1411196319 

Another instance of this issue. Scrollbar should be dark. There are a few more elements that are rendered outside the root element and all will have this issue. 

1. Popovers. 
2. Tooltips. 
3. Modal.
4. Might be more.. can't remember.


Screenshot 2023-12-05 at 2 04 02 AM


**Should be like....**

Screenshot 2023-12-05 at 2 02 24 AM

**Comment by grgia** [comment]
@chrispader @parasharrajat maybe we should tackle the web scrollbar color changes in a separate PR and remove that change from this PR so we can finalize the migration?

**Comment by grgia** [comment]
@parasharrajat, although I'm unable to replicate that on this branch on DEV (setting my computer to system light mode and app in dark mode works fine). Are your screenshots from staging?

**Comment by chrispader** [comment]
> @parasharrajat, although I'm unable to replicate that on this branch on DEV (setting my computer to system light mode and app in dark mode works fine). Are your screenshots from staging?

i also can't reproduce this @parasharrajat 

> [#32259 (comment)](https://github.com/Expensify/App/pull/32259#discussion_r1411196319)
> 
> Another instance of this issue. Scrollbar should be dark. There are a few more elements that are rendered outside the root element and all will have this issue.

Fixed this! Can't find any other cases where we there might be scrollable content outside of `:root`

**Review by parasharrajat — APPROVED** [review]
## Reviewer Checklist

- [x] I have verified the author checklist is complete (all boxes are checked off).
- [x] I verified the correct issue is linked in the `### Fixed Issues` section above
- [x] I verified testing steps are clear and they cover the changes made in this PR
    - [x] I verified the steps for local testing are in the `Tests` section
    - [x] I verified the steps for Staging and/or Production testing are in the `QA steps` section
    - [x] I verified the steps cover any possible failure scenarios (i.e. verify an input displays the correct error message if the entered data is not correct)
    - [x] I turned off my network connection and tested it while offline to ensure it matches the expected behavior (i.e. verify the default avatar icon is displayed if app is offline)
- [x] I checked that screenshots or videos are included for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [x] I included screenshots or videos for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [x] I verified tests pass on **all platforms** & I tested again on:
    - [x] Android / native
    - [x] Android / Chrome
    - [x] iOS / native
    - [x] iOS / Safari
    - [x] MacOS / Chrome / Safari
    - [x] MacOS / Desktop
- [x] If there are any errors in the console that are unrelated to this PR, I either fixed them (preferred) or linked to where I reported them in Slack
- [x] I verified proper code patterns were followed (see [Reviewing the code](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md#reviewing-the-code))
    - [x] I verified that any callback methods that were added or modified are named for what the method does and never what callback they handle (i.e. `toggleReport` and not `onIconClick`).
    - [x] I verified that the left part of a conditional rendering a React component is a boolean and NOT a string, e.g. `myBool && <MyComponent />`.
    - [x] I verified that comments were added to code that is not self explanatory
    - [x] I verified that any new or modified comments were clear, correct English, and explained "why" the code was doing something instead of only explaining "what" the code was doing.
    - [x] I verified any copy / text shown in the product is localized by adding it to `src/languages/*` files and using the [translation method](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60)
    - [x] I verified all numbers, amounts, dates and phone numbers shown in the product are using the [localization methods](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60-L68)
    - [x] I verified any copy / text that was added to the app is grammatically correct in English. It adheres to proper capitalization guidelines (note: only the first word of header/labels should be capitalized), and is approved by marketing by adding the `Waiting for Copy` label for a copy review on the original GH to get the correct copy.
    - [x] I verified proper file naming conventions were followed for any new files or renamed files. All non-platform specific files are named after what they export and are not named "index.js". All platform-specific files are named for the platform the code supports as outlined in the README.
    - [x] I verified the JSDocs style guidelines (in [`STYLE.md`](https://github.com/Expensify/App/blob/main/contributingGuides/STYLE.md#jsdocs)) were followed
- [x] If a new code pattern is added I verified it was agreed to be used by multiple Expensify engineers
- [x] I verified that this PR follows the guidelines as stated in the [Review Guidelines](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md)
- [x] I verified other components that can be impacted by these changes have been tested, and I retested again (i.e. if the PR modifies a shared library or component like `Avatar`, I verified the components using `Avatar` have been tested & I retested again)
- [x] I verified all code is DRY (the PR doesn't include any logic written more than once, with the exception of tests)
- [x] I verified any variables that can be defined as constants (ie. in CONST.js or at the top of the file that uses the constant) are defined as such
- [x] If a new component is created I verified that:
    - [x] A similar component doesn't exist in the codebase
    - [x] All props are defined accurately and each prop has a `/** comment above it */`
    - [x] The file is named correctly
    - [x] The component has a clear name that is non-ambiguous and the purpose of the component can be inferred from the name alone
    - [x] The only data being stored in the state is data necessary for rendering and nothing else
    - [x] For Class Components, any internal methods passed to components event handlers are bound to `this` properly so there are no scoping issues (i.e. for `onClick={this.submit}` the method `this.submit` should be bound to `this` in the constructor)
    - [x] Any internal methods bound to `this` are necessary to be bound (i.e. avoid `this.submit = this.submit.bind(this);` if `this.submit` is never passed to a component event handler like `onClick`)
    - [x] All JSX used for rendering exists in the render method
    - [x] The component has the minimum amount of code necessary for its purpose, and it is broken down into smaller components in order to separate concerns and functions
- [x] If any new file was added I verified that:
    - [x] The file has a description of what it does and/or why is needed at the top of the file if the code is not self explanatory
- [x] If a new CSS style is added I verified that:
    - [x] A similar style doesn't already exist
    - [x] The style can't be created with an existing [StyleUtils](https://github.com/Expensify/App/blob/main/src/styles/StyleUtils.js) function (i.e. `StyleUtils.getBackgroundAndBorderStyle(themeColors.componentBG`)
- [x] If the PR modifies code that runs when editing or sending messages, I tested and verified there is no unexpected behavior for all supported markdown - URLs, single line code, code blocks, quotes, headings, bold, strikethrough, and italic.
- [x] If the PR modifies a generic component, I tested and verified that those changes do not break usages of that component in the rest of the App (i.e. if a shared library or component like `Avatar` is modified, I verified that `Avatar` is working as expected in all cases)
- [x] If the PR modifies a component related to any of the existing Storybook stories, I tested and verified all stories for that component are still working as expected.
- [x] If a new page is added, I verified it's using the `ScrollView` component to make it scrollable when more elements are added to the page.
- [x] If the `main` branch was merged into this PR after a review, I tested again and verified the outcome was still expected according to the `Test` steps.
- [x] I have checked off every checkbox in the PR reviewer checklist, including those that don't apply to this PR.


:ribbon: :eyes: :ribbon:  C+ reviewed

**Review by grgia — APPROVED** [review]

**Pull request merged**